framework/src/modules/token/stores/supported_tokens.ts

```diff
@@ -0,0 +1,137 @@
+};
+
+export const ALL_SUPPORTED_TOKENS_KEY = Buffer.from([0, 0, 0, 0]);
+
+export class SupportedTokensStore extends BaseStore<SupportedTokensStoreData> {
+	public schema = supportedTokensStoreSchema;
+
+	private _ownChainID!: Buffer;
+
+	public registerOwnChainID(ownChainID: Buffer) {
+		this._ownChainID = ownChainID;
+	}
+
+	public async allSupported(context: ImmutableStoreGetter): Promise<boolean> {
+		const allSupportedTokens = await this.iterate(context, {
+			gte: Buffer.alloc(4, 0),
+			lte: Buffer.alloc(4, 255),
+		});
+
+		return (
+			allSupportedTokens.length === 1 && allSupportedTokens[0].key.equals(ALL_SUPPORTED_TOKENS_KEY)
+		);
+	}
+
+	public async isSupported(context: ImmutableStoreGetter, tokenID: Buffer): Promise<boolean> {
+		if (this._isMainchainOrNative(tokenID)) {
+			return true;
+		}
+		const allSupported = await this.allSupported(context);
+		if (allSupported) {
+			return true;
+		}
+		const chainID = tokenID.slice(0, CHAIN_ID_LENGTH);
+		try {
+			const supported = await this.get(context, chainID);
+			if (
+				supported.supportedTokenIDs.length === 0 ||
+				supported.supportedTokenIDs.findIndex(id => id.equals(tokenID)) > -1
+			) {
+				return true;
+			}
+		} catch (error) {
+			if (!(error instanceof NotFoundError)) {
+				throw error;
+			}
+		}
+		return false;
+	}
+
+	public async supportAll(context: StoreGetter): Promise<void> {
+		const allSupportedTokens = await this.iterate(context, {
+			gte: Buffer.alloc(4, 0),
+			lte: Buffer.alloc(4, 255),
+		});
+		for (const { key } of allSupportedTokens) {
+			await this.del(context, key);
+		}
+		await this.set(context, ALL_SUPPORTED_TOKENS_KEY, { supportedTokenIDs: [] });
+	}
+
+	public async supportChain(context: StoreGetter, chainID: Buffer): Promise<void> {
+		// LSK or native token do not need entry
+		if (this._isMainchainOrNative(Buffer.concat([chainID, Buffer.alloc(LOCAL_ID_LENGTH)]))) {
+			return;
+		}
+		await this.del(context, ALL_SUPPORTED_TOKENS_KEY);
+		await this.set(context, chainID, { supportedTokenIDs: [] });
+	}
+
+	public async supportToken(context: StoreGetter, tokenID: Buffer): Promise<void> {
+		// LSK or native token do not need entry
+		if (this._isMainchainOrNative(tokenID)) {
+			return;
+		}
+		const chainID = tokenID.slice(0, CHAIN_ID_LENGTH);
+		let supported: SupportedTokensStoreData = { supportedTokenIDs: [] };
+		try {
+			supported = await this.get(context, chainID);
+		} catch (error) {
+			if (!(error instanceof NotFoundError)) {
+				throw error;
+			}
+		}
+		supported.supportedTokenIDs.push(tokenID);
+		supported.supportedTokenIDs.sort((a, b) => a.compare(b));
+		await this.del(context, ALL_SUPPORTED_TOKENS_KEY);
+		await this.set(context, chainID, supported);
+	}
+
+	private _isMainchainOrNative(tokenID: Buffer): boolean {
+		const [chainID] = splitTokenID(tokenID);
+		if (chainID.equals(this._ownChainID)) {
+			return true;
+		}
+		if (chainID[0] === this._ownChainID[0] && chainID.slice(1).equals(CHAIN_ID_LSK.slice(1))) {
```

Comment:
What do you think of replacing lines 132 - L135 with 
`return chainID[0] === this._ownChainID[0] && chainID.slice(1).equals(CHAIN_ID_LSK.slice(1));`